src/mqueryfront/src/components/StatusProgress.js

```diff
@@ -1,35 +1,14 @@
 import React from "react";
 import PropTypes from "prop-types";
+import { getProgressBarClass } from "../queryUtils";
 
 const StatusProgress = (props) => {
-    let rowClass;
-    switch (props.status) {
-        case "done":
-            rowClass = "success";
-            break;
-        case "processing":
-            rowClass = "info";
-            break;
-        case "querying":
-            rowClass = "info";
-            break;
-        case "cancelled":
-            rowClass = "danger";
-            break;
-        case "expired":
-            rowClass = "warning";
-            break;
-        default:
-            rowClass = "";
-            break;
-    }
+    const { status, total_files, files_processed } = props;
 
-    const progressClass = "progress-bar bg-" + rowClass;
+    const progressClass = getProgressBarClass(status);
 
     const percentage = Math.round(
-        props.total_files
-            ? (props.files_processed * 100) / props.total_files
-            : 0
+        total_files ? (files_processed * 100) / total_files : 0
```

Comment:
There's a small bug here btw, and maybe it's a good time for fix it:

when the status is finished (for example, `failed`, 0/0 should render as completely filled in).

For example, `done` job that matched 0 files should be 100% green instead of empty.
The same goes for `failed` job that errored during parsing.

So instead of `0`, maybe it should be something like `isStatusFinished(status) ? 100 : 0`

Reply:
OK. I pass isFinished flag by props to component. After this change, it looks better.